### ISSUE TITLE
[Bug] [dolphinscheduler-master] SERIAL_WAIT command_param does not replace global_params, rendering the parameters passed via the API ineffective

### ISSUE THREAD

**Issue by 1205747671** [apache/dolphinscheduler#14566]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When I run multiple instances of the same workflow using the API and the execution strategy of the workflow definition is set to serial wait, the state of the workflow instances changes from "state":"SERIAL_WAIT","desc":"wait by serial_wait strategy" to "state":"RUNNING_EXECUTION","desc":"recover serial wait". In this case, the start params passed in through the API does not replace the global_params, which results in the parameters passed through the API becoming ineffective.

API URL: projects/{projectCode}/executors/start-process-instance

### What you expected to happen

The parameters passed through the API should properly replace the global_params.

### How to reproduce

Running multiple instances at once through API.

### Anything else

I rapidly ran multiple instances of the same workflow through the API with the execution policy established as serial waiting. In theory, these instances should execute sequentially. However, I've observed that some instances are executing simultaneously instead of following the expected waiting sequence.

### Version

3.1.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by CallMeKingsley97** [comment]
I'll try to fix it

**Comment by 1205747671** [comment]
> I'll try to fix it

I am using the API "projects/{projectCode}/executors/start-process-instance" to quickly run multiple instances of the same serial waiting workflow, such as in a for loop calling this interface. In theory, these instances should be executed in order. However, I observed that some instances are executed simultaneously, rather than in the expected waiting order. Do I need to add another issue for this? I have already mentioned this above, but I see you haven't resolved it yet.

**Comment by CallMeKingsley97** [comment]
> 

Oh, sorry, I forgot it. Please add another issue for this problem.😂
